### ISSUE TITLE
Allow manually setting a buffer language

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/community#207]
In VSCode you can manually set the language of a buffer:

![](https://user-images.githubusercontent.com/1714999/164590322-635dea49-4aba-42ac-88c6-6f863484d236.png)

This is especially useful for new/unsaved files that won't ever be saved.

**Comment by gaetgu** [comment]
:+1:

**Comment by aiakared** [comment]
An option in `settings.json` similar to VS Code's `files.associations` where you can map a file extension to a language would be helpful too for file types that aren't yet supported but could be partially supported by changing the language mode, e.g:
```json
{
    "language_overrides": {
        ".vue": "html",
    }
}
```

**Comment by chriskrycho** [comment]
Seconded; this is particularly handy for working with something like HTML templating languages while support is still in progress.

**Comment by chloerei** [comment]
I want `.html.erb` `.turbo_stream.erb` support html tag and attribute completion.

**Comment by JosephTLyons** [comment]
This should be available in v0.76.1.  Note, this does not introduce any project-specific settings, it is just a popup menu that allows you to choose the language for the given file.
